manufacturing-client/src/main.rs

```diff
@@ -291,15 +293,57 @@ async fn main() -> Result<()> {
 }
 
 async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType> {
-    if let Some(mfg_info) = env::var_os("MANUFACTURING_INFO") {
-        return Ok(CborSimpleType::Text(mfg_info.into_string().unwrap()));
-    }
     match mfg_string_type {
         MfgStringType::SerialNumber => {
-            let serial = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
+            let serial_dmi = fs::read_to_string("/sys/devices/virtual/dmi/id/product_serial")
                 .or_else(|_| fs::read_to_string("/sys/devices/virtual/dmi/id/chassis_serial"))
                 .context("Error determining system serial number")?;
-            Ok(CborSimpleType::Text(serial))
+            let mut arg: String;
+            let output_ifaces = Command::new("ls")
+                .arg("/sys/class/net/")
+                .stdout(std::process::Stdio::piped())
+                .stderr(std::process::Stdio::piped())
+                .output()
+                .expect("Failed to execute command");
+
+            let val = String::from_utf8_lossy(&output_ifaces.stdout).to_string();
+            let mut serial_mac: String = "".to_string();
+            for interface in val.split('\n') {
+                if interface.starts_with('e') || interface.starts_with('w') {
```

Comment:
I like that option, it will be better way to go through all network interfaces without concerning the name (even if it changes by some means)and get proper mac address.

Reply:
yeah, mac(s) can be changed and I agree we should just discard empty mac and loopback

Reply:
added check for valid and non-zero MAC address in latest commit . so that it will drop empty mac